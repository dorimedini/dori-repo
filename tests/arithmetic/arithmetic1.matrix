{
	matrix m_square(3,3) = [1, 2, 3; 4, 5, 6; 7, 8, 9];
	matrix m(2,3) = [1, 2, 3; 4, 5, 6];
	print m_square;
	print m;

	foreach (i in m_square)
		print i;

	matrix trans(3,2)=transpose(m), trans_sq=transpose(m_square);
	matrix I(3,3) = [1, 0, 0; 0, 1, 0; 0, 0, 1];

	matrix add(3,3)=m_square + I;
	print add;
	m_square=m_square+I;
	print m_square;
	m_square=m_square-I;
	print m_square;

	matrix subtract(3,3)=m_square - I;
	print subtract;
	m_square=m_square-I;
	print m_square;
	m_square=m_square+I;
	print m_square;

	matrix mul_id(3,3)=m_square * I;
	print mul;
	
	m_square=m_square * m_square;
	print m_square;
	
	matrix mul_trans(2,2)=m * transpose(m);
	print mul_trans;

	matrix neg(2,3)=-m;
	print neg;

	int a = 3;

	m=m * a;
	print m;
	int x=0,y=0;
	foreach(i in m) {
		m[x,y]=m[x,y]/3;
		x=x+1;
		y=y+1;
	}
	print m;
	
	m=a*m;
	print m;
	int x=0,y=0;
	foreach(i in m) {
		m[x,y]=m[x,y]/3;
		x=x+1;
		y=y+1;
	}
	print m;
	
	m=m*-1;
	print m;
	int x=0,y=0;
	foreach(i in m) {
		m[x,y]=m[x,y]/3;
		x=x+1;
		y=y+1;
	}
	print m;
	
	m=m*(-1);
	print m;
	int x=0,y=0;
	foreach(i in m) {
		m[x,y]=m[x,y]/3;
		x=x+1;
		y=y+1;
	}
	print m;
	
	m=m*0;
	print m;
	int x=0,y=0;
	foreach(i in m) {
		m[x,y]=m[x,y]/3;
		x=x+1;
		y=y+1;
	}
	print m;
	
}