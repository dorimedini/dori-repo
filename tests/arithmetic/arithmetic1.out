[1,2,3;4,5,6;7,8,9]
[1,2,3;4,5,6]
1
2
3
4
5
6
7
8
9
[2,2,3;4,6,6;7,8,10]
[2,2,3;4,6,6;7,8,10]
[1,2,3;4,5,6;7,8,9]
[0,2,3;4,4,6;7,8,8]
[0,2,3;4,4,6;7,8,8]
[1,2,3;4,5,6;7,8,9]
[1,2,3;4,5,6;7,8,9]
[30,36,42;66,81,96;102,126,150]
[14,32;32,77]
[-1,-2,-3;-4,-5,-6]
[3,6,9;12,15,18]
[1,2,3;4,5,6]
[3,6,9;12,15,18]
[1,2,3;4,5,6]
[-1,-2,-3;-4,-5,-6]
[1,2,3;4,5,6]
[-1,-2,-3;-4,-5,-6]
[1,2,3;4,5,6]
[0,0,0;0,0,0]
[0,0,0;0,0,0]
